File: Extensions.Tests/SqueezeTests.fs
namespace Tests.Extensions

open System
open Stringier.Extensions
open Microsoft.VisualStudio.TestTools.UnitTesting

[<TestClass>]
type SqueezeTests () =
    [<DataTestMethod>]
    [<DataRow("helo", "hello")>]
    [<DataRow("Leroy Jenkins", "Leeeeeeeerooooooooooyyyyyyyy Jeeeeeeeenkiiiiiiiiins")>]
    member _.``squeeze string`` (exp:string, src:string) =
        Assert.AreEqual(exp, squeeze src)

File: Tests/SqueezeTests.cs
using System;
using Stringier;
using Defender;
using Xunit;

namespace Tests {
	public class SqueezeTests : Trial {
		[Theory]
		[InlineData("hello", "helo")]
		[InlineData("Leeeeeeeerooooooooooyyyyyyyy Jeeeeeeeenkiiiiiiiiins", "Leroy Jenkins")]
		public void Squeeze(String source, String expected) => Claim.That(source.Squeeze()).Equals(expected);
	}
}
